File: solutions/java/house/1/src/main/java/House.java
import java.util.stream.Collectors;
import java.util.stream.IntStream;

class House {
    public String verse(int verseNumber) {
        if (verseNumber < 1 || verseNumber > 12)
            return "";

        String result = "This is ";

        switch (verseNumber) {
            case 12:
                result += "the horse and the hound and the horn that belonged to ";
            case 11:
                result += "the farmer sowing his corn that kept ";
            case 10:
                result += "the rooster that crowed in the morn that woke ";
            case 9:
                result += "the priest all shaven and shorn that married ";
            case 8:
                result += "the man all tattered and torn that kissed ";
            case 7:
                result += "the maiden all forlorn that milked ";
            case 6:
                result += "the cow with the crumpled horn that tossed ";
            case 5:
                result += "the dog that worried ";
            case 4:
                result += "the cat that killed ";
            case 3:
                result += "the rat that ate ";
            case 2:
                result += "the malt that lay in ";
            case 1:
                result += "the house that Jack built.";
        }
        return result;
    }

    public String verses(int startVerse, int endVerse) {
        return IntStream.range(startVerse, endVerse + 1).mapToObj(i -> this.verse(i)).collect(Collectors.joining("\n"));
    }

    public String sing() {
        return this.verses(1, 12);
    }
}

File: solutions/python/house/1/house.py
def recite(start_verse: int, end_verse: int) -> list[str]:
    verses = ["the house", 
              "the malt that lay in",
              "the rat that ate",
              "the cat that killed",
              "the dog that worried",
              "the cow with the crumpled horn that tossed",
              "the maiden all forlorn that milked",
              "the man all tattered and torn that kissed",
              "the priest all shaven and shorn that married",
              "the rooster that crowed in the morn that woke",
              "the farmer sowing his corn that kept",
              "the horse and the hound and the horn that belonged to"
             ]

    return ["This is " + " ".join(reversed(verses[0:i]))+ " that Jack built." for i in range(start_verse, end_verse + 1)]
